projects/vdk-core/src/vdk/api/plugin/plugin_input.py

```diff
@@ -317,21 +327,62 @@ def pre_ingest_process(self, payload: List[dict]) -> List[dict]:
         :param metadata: Optional[IngestionMetadata] dict object containing
         metadata produced by the pre-ingest plugin, and possibly used by
         other pre-ingest, ingest, or post-ingest plugins.
+            NOTE: A read-only parameter. Whatever modifications are done to
+            this object, once returned, it is treated as a new object.
+        :param destination_table: Optional[string]
+            (Optional) The name of the table, where the data should be
+            ingested into. This argument is optional, and needs to be
+            considered only when the name of the destination table is not
+            included in the payload itself.
+            NOTE: A read-only parameter. It is not expected to be modified and
+            returned.
+        :param target: Optional[string]
+            (Optional) Used to identify where the data should be ingested into.
+                Specifies a data source and its destination database.
+                The values for this parameter can be in the format
+                `<some-data-source_and-db-table>`, or as a URL.
+                Example:
+                      http://example.com/<some-api>/<data-source_and-db-table>
+            This parameter does not need to be used, in case the
+            `INGEST_TARGET_DEFAULT` environment variable is set. This can be
+            made by plugins, which may set default value, or it can be
+            overwritten by users.
+            NOTE: A read-only parameter. It is not expected to be modified and
+            returned.
+        :param collection_id: string
+            (Optional) An identifier to indicate that data from different
+            method invocations belong to same collection. Defaults to
+            "data_job_name|OpID", meaning all method invocations from a data
+            job run will belong to the same collection.
+            NOTE: A read-only parameter. It is not expected to be modified and
+            returned.
         :return: (List[dict], Optional[IngestionMetadata]), a tuple
-        containing the processed payload and possibly an IngestionMetadata
-        object.
+            containing the processed payload and possibly an IngestionMetadata
+            object.
+
+        :exception: If an exception occurs during this operation, all other
+            pre-processing operations and the ingestion of the payload would
+            be terminated to prevent data corruption. In case the ingestion is
+            expected to happen regardless if there are exceptions or not,
+            then the exception handling needs to be implemented within this
+            method.
         """
         pass
 
     def post_ingest_process(
         self,
         payload: Optional[List[dict]] = None,
         ingestion_metadata: Optional[IngestionMetadata] = None,
-        exceptions: Optional[List] = None,
-    ) -> None:
+        exception: Optional[Exception] = None,
```

Comment:
Would rather move exception param to last, since it is only passed for exceptional cases therefore rare.
When a user reviews the interface surface, the common to ingest_process() parameters in the known order would be easier to spot, then the newly added parameters.

Reply:
Refactored for consistency. Thanks